src/src/com/microsoft/aad/adal/AuthenticationContext.java

```diff
@@ -1153,6 +1153,16 @@ private AuthenticationResult acquireTokenAfterValidation(CallbackHandler callbac
             AuthenticationResult result = null;
             request.setVersion(getVersionName());
             request.setBrokerAccountName(request.getLoginHint());
+            
+            //check the redirectUri before sending the request
+            //if the redirectUri from the client does not match the valid redirectUri
+            // the client app would not jump to the login page
```

Comment:
This is under the case canSwithToBroker=true. But this code looks different compared to another PR for checking redirect uri @heidijinxujia .